### PR TITLE
section 5 rule 40

### PR DESCRIPTION
**Pull request by weilixu**
Not finalized - the RDS seems unclear on two places:
1. Applicability check: building level or surface level?
2. Door sub surface classification type - this [Enum ](https://github.com/open229/ruleset-model-description-schema/blob/main/docs229/Enumerations2019ASHRAE901.schema.json#L1020)contains more values than door classifications.

### CODE REVIEW

**Comment by weilixu** [comment]
Based on RDS discussion:
door subclassification -> if subclassification is `SPANDREL_GLASS`, `GLASS_BLOCK`, `OTHER`, then the rule shall be undetermined.

The applicability check shall happen at building level as well as each component level.

File: rct229/rulesets/ashrae9012019/section5/section5rule53.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+            def list_filter(self, context_item, data):
+                subsurface_b = context_item.baseline
+                # exclude non door surfaces and doors whose opaque area is greater than glazed area
+                return (
+                    subsurface_b["classification"] == DOOR
+                    and subsurface_b["glazed_area"] >= subsurface_b["opaque_area"]
+                )
+
+            class SubSurfaceRule(RuleDefinitionBase):
```
Comment (charliepnnl) [inline rct229/rulesets/ashrae9012019/section5/section5rule53.py:122]: The implementation doesn't seem to match the RDS. Can the RDS be updated?

**Review by charliepnnl — COMMENTED** [review]
I can't fully review this since the implementation doesn't match the RDS. From the discussion, it seems that the logic has be changed but the RDS hasn't been updated to match.  Can the RDS be updated?

File: rct229/rulesets/ashrae9012019/section5/section5rule53.py (hunk 1 above)
Comment (JacksonJ-KC) [reply rct229/rulesets/ashrae9012019/section5/section5rule53.py:122]: The RDS is now updated and approved

File: rct229/rulesets/ashrae9012019/section5/section5rule53.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                    >= subsurface_b.get("opaque_area", ZERO.AREA)
+                    for subsurface_b in subsurfaces
+                ]
+            )
+
+        def create_data(self, context, data=None):
+            building_b = context.baseline
+            climate_zone = data["climate_zone"]
+            return {
+                "surface_conditioning_category_dict_b": get_surface_conditioning_category_dict(
```
[the remark this answers is not in the document]
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section5/section5rule53.py:76]: The updated RDS fixed this issue.

File: rct229/rulesets/ashrae9012019/section5/section5rule53.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+                    climate_zone, building_b
+                )
+            }
+
+        def list_filter(self, context_item, data=None):
+            # context_item shall be the list of the list_path element
+            surface_b = context_item.baseline
+            # exclude unregulated surfaces
+            return (
+                data["surface_conditioning_category_dict_b"][surface_b["id"]]
```
[the remark this answers is not in the document]
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section5/section5rule53.py:86]: The updated RDS has this part now. However, ```if (scc_b == "UNREGULATED"):``` should be ```if (scc_b != "UNREGULATED"):```.

**Comment by weilixu** [comment]
This Rule is renumbered to 5-40

File: rct229/rulesets/ashrae9012019/section5/section5rule53.py [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+                    )
+
+                    target_u_factor_res = table_G34_lookup(
+                        climate_zone,
+                        SCC.EXTERIOR_RESIDENTIAL,
+                        DOOR,
+                        subsurface_b["subclassification"],
+                    )
+
+                    return (
```
[the remark this answers is not in the document]
Comment (weilixu) [reply rct229/rulesets/ashrae9012019/section5/section5rule53.py:154]: This is developed almost a year ago. Does it block your progress? I can look into it this week.

[the remark this answers is not in the document]
Comment (JacksonJ-KC) [reply rct229/rulesets/ashrae9012019/section5/section5rule53.py:154]: I can develop test cases based on the RDS and wait until this implementation is reviewed/updated to test them

File: rct229/rulesets/ashrae9012019/data_fns/table_G3_4_fns.py [hunk 5]
```diff
@@ -154,6 +165,9 @@ def table_G34_lookup(
     if skylit_wwr is not None:  # when `skylit_wwr ==0`, `if skylit_wwr:` becomes False
         search_criteria.extend(skylit_to_search_criteria(skylit_wwr))
 
+    if classification is not None:
+        search_criteria.append(("standards_construction_type", classification))
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/data_fns/table_G3_4_fns.py:169]:
```python
search_criteria.append(("standards_construction_type", DOOR_SUBCLASSIFICATION_TYPE_TO_STANDARD_CONSTRUCTION_MAP[classification]))
```

File: rct229/rulesets/ashrae9012019/section5/section5rule53.py [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_surface_conditioning_category_dict import (
+    get_surface_conditioning_category_dict,
+)
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.pint_utils import ZERO
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_surface_conditioning_category_dict import (
+    SurfaceConditioningCategory as SCC,
+)
+
+DOOR = schema_enums["SubsurfaceClassificationOptions"].DOOR
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section5/section5rule53.py:15]:
```python
SPANDREL_GLASS = schema_enums.schema_enums["SubsurfaceSubclassificationOptions2019ASHRAE901"].SPANDREL_GLASS 
GLASS_BLOCK = schema_enums.schema_enums["SubsurfaceSubclassificationOptions2019ASHRAE901"].GLASS_BLOCK 
OTHER = schema_enums.schema_enums["SubsurfaceSubclassificationOptions2019ASHRAE901"].OTHER
```

File: rct229/rulesets/ashrae9012019/section5/section5rule53.py [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+                        Section5Rule53.BuildingRule.SurfaceRule.SubSurfaceRule, self
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, False),
+                        required_fields={
+                            "$": ["subclassification", "u_factor"],
+                        },
+                        manual_check_required_msg=MANUAL_CHECK_MSG,
+                    )
+
+                def is_applicable(self, context, data=None):
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section5/section5rule53.py:134]: Revise this to undetermined.

File: rct229/rulesets/ashrae9012019/section5/section5rule53.py [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+                        DOOR,
+                        subsurface_b["subclassification"],
+                    )
+
+                    return (
+                        surface_conditioning_category_b == SCC.EXTERIOR_MIXED
+                        and target_u_factor_nonres != target_u_factor_res
+                    )
+
+                def get_manual_check_required_msg(
```
[the remark this answers is not in the document]
Comment (weilixu) [reply rct229/rulesets/ashrae9012019/section5/section5rule53.py:159]: No, the `is_applicable` shall be the `manual_check_required`.

**Comment by weilixu** [comment]
This PR is addressed in #1248. Close this PR.

**Pull request closed without merging**